### ISSUE TITLE
Move gateway unit tests inside the same folder of test target

### ISSUE THREAD

**Issue by kenfdev** [openfaas/faas#625]
Currently some of the gateway unit tests are inside `gateway/tests` directory (e.g. `gateway/tests/add_metrics_test.go`) . We should move these unit tests inside the same directory as the test target (e.g. `gateway/metrics/add_metrics.go` ). The coverage will show better percentages and editors could highlight the test covered areas.

## Expected Behaviour

The unit tests should take effect to the coverage when they are executed.

e.g.
`gateway/tests/add_metrics_test.go`
move to
`gateway/metrics/add_metrics_test.go`
which is the same directory as
`gateway/metrics/add_metrics.go`
```
gateway git:(master) ✗ go test $(go list ./... | grep -v integration | grep -v /vendor/ | grep -v /template/) -cover
?       github.com/openfaas/faas/gateway        [no test files]
ok      github.com/openfaas/faas/gateway/handlers       0.021s  coverage: 24.9% of statements
ok      github.com/openfaas/faas/gateway/metrics        0.020s  coverage: 30.2% of statements
?       github.com/openfaas/faas/gateway/plugin [no test files]
?       github.com/openfaas/faas/gateway/queue  [no test files]
ok      github.com/openfaas/faas/gateway/requests       0.022s  coverage: 75.0% of statements
ok      github.com/openfaas/faas/gateway/types  0.018s  coverage: 81.2% of statements
```

## Current Behaviour

The coverage isn't affected by the unit tests inside the `gateway/tests` directory hence coverage doesn't look good.

```
gateway git:(master) go test $(go list ./... | grep -v integration | grep -v /vendor/ | grep -v /template/) -cover
?       github.com/openfaas/faas/gateway        [no test files]
ok      github.com/openfaas/faas/gateway/handlers       0.022s  coverage: 16.0% of statements
?       github.com/openfaas/faas/gateway/metrics        [no test files]
?       github.com/openfaas/faas/gateway/plugin [no test files]
?       github.com/openfaas/faas/gateway/queue  [no test files]
?       github.com/openfaas/faas/gateway/requests       [no test files]
ok      github.com/openfaas/faas/gateway/tests  0.026s  coverage: 0.0% of statements
?       github.com/openfaas/faas/gateway/types  [no test files]
```

## Possible Solution

Move the unit tests inside the same directory as the test target.

## Steps to Reproduce (for bugs)

1. `cd` to the `gateway` directory
2. `go test $(go list ./... | grep -v integration | grep -v /vendor/ | grep -v /template/) -cover`

## Your Environment

* Operating System and version (e.g. Linux, Windows, MacOS):
MacOS

* Please also follow the [troubleshooting guide](https://github.com/openfaas/faas/blob/master/guide/troubleshooting.md) and paste in any other diagnostic information you have:

**Comment by kenfdev** [comment]
Derek add label: proposal

**Comment by kenfdev** [comment]
I can create a PR on this tomorrow.

**Comment by alexellis** [comment]
Hi Ken thanks for raising this issue. Please go ahead and move the test files in a PR.

**Comment by alexellis** [comment]
Fixed

**Comment by alexellis** [comment]
Ken FYI in future you can add a comment like Fixes #625 then after the merge the issue will be automatically closed also.

**Comment by kenfdev** [comment]
ahh.. Thanks for letting me know. Will do from now on
